bool enable
float64 init
float64 goal
